File: RAI/CatExample/PythonImplementation/Cat.py
class Cat:
    #name:str

    def __init__(self, name, sex, age, weight, color, texture):
        self._name = name
        self.sex = sex
        self.age = age
        self.weight = weight
        self.__color = color
        self.texture = texture

    def get_name(self)->str:
        return self._name

    def get_color(self)->str:
        return self.__color

    def set_color(self, nuevo_color:str)->bool:
        colores_validos = ["blanco","gris","negro","naranja","marron"]
        for color in colores_validos:
            if color == nuevo_color:
                self.__color = nuevo_color
                return True
        return False

    def meow(self):
        print(self._name + " hace MEOWWW!!!")

File: RAI/CatExample/JavaImplementation/Cat.java
public class Cat{
    //Atributos
    private String name;
    private String sex;
    private int age;
    private int weight;
    private String color;
    private String textura;

    //Métodos
    public Cat(String nombre, String sex, int age, int weight, String color, String textura){
        this.name = nombre;
        this.sex = sex;
        this.age = age;
        this.weight = weight;
        this.color = color;
        this.textura = textura;
    }

    public String getName(){
        return this.name;
    }

    public String getColor(){
        return this.color;
    }

    public void setColor(String nuevoColor){
        String[] coloresValidos = {"blanco","marron","naranja","negro","gris"};
        if(colorEsValido(coloresValidos, nuevoColor)){
            this.color = nuevoColor;
        }
    }

    private boolean colorEsValido(String[] colores, String nuevoColor){
        for(String color:colores){
            System.out.println("color: " +color+ " y nuevoColor: " + nuevoColor);
            if(color.compareTo(nuevoColor)==0){
                return true;
            }
        }
        return false;
    }

    public void meow(){
        System.out.println(this.name + " hace MEOWWWW!!!");
    }
}